source/common/http/header_utility.cc

```diff
@@ -216,6 +216,27 @@ bool HeaderUtility::isEnvoyInternalRequest(const RequestHeaderMap& headers) {
          internal_request_header->value() == Headers::get().EnvoyInternalRequestValues.True;
 }
 
+void HeaderUtility::stripTrailingHostDot(RequestHeaderMap& headers) {
+  auto host = headers.getHostValue();
+  // If the host ends in a period, remove it.
+  auto dot_index = host.rfind('.');
+  if (dot_index == std::string::npos) {
+    return;
+  } else if (dot_index == (host.size() - 1)) {
+    host.remove_suffix(1);
+    headers.setHost(host);
+    return;
+  }
+  // If the dot is just before a colon, it must be preceding the port number.
+  // Because although the host may contain a colon via an IPv6 bracketed
+  // address, and although that IPv6 address may also contain dots when
+  // embedding an address per RFC 4291 2.2.3. In this case the dot will never
```

Comment:
This is a bit tricky to read, maybe something like `IPv6 addresses may contain colons or dots, but the dot will never directly precede the colon, so this check should be sufficient to detect a trailing port number.`?

Reply:
sure, will update it.